tests/Microsoft.Bot.Builder.Tests/BotFrameworkAdapterOAuthTests.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.using System.Security.Claims;
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using Moq.Protected;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.Tests
+{
+    [TestClass]
+    public class BotFrameworkAdapterOAuthTests
+    {
+        [TestMethod]
+        public async Task BotFrameworkAdapterFindsOAuthCards()
```

Comment:
A single test for all the possibilities that your code is introducing seems insufficient for me.

For example, you had lines that had 3 question marks, that means that 3 things could be null or none be null. That is 4 cases. 

OTher relevant tests:

-Polling until timeout: do things end correctly?
-Polling and getting the token first try
-Polling and getting the token second try 

Can this be tested at all? Seems more like functional tests but if they could be done that would be great. We can add secrets in the build system if you need any, let me know, happy to help here.

Reply:
I added more test cases. It can be tested with all the Mocks I made for the ConnectorClient and OAuthClient.